### PR TITLE
German Oil Temperatures

### PR DESCRIPTION
**Pull request by itsjustdel**

### CODE REVIEW

File: Il2ServerAppLateNight/Il2ServerAppLateNight/OilTemp.cpp [hunk 1]
```diff
@@ -13,6 +13,70 @@ bool Intake(std::string planeName)
...
+
+bool isJU52(std::string planeName)
+{
+	std::string v = "Ju-52/3m g4e";
+	if (planeName.compare(v) == 0)
+		return true;
+
+	return false;
+}
+
```
Comment (itsjustdel) [inline Il2ServerAppLateNight/Il2ServerAppLateNight/OilTemp.cpp:78]: These could all be moved to a helper function

File: Il2ServerAppLateNight/Il2ServerAppLateNight/OilTemp.cpp [hunk 2]
```diff
@@ -13,6 +13,70 @@ bool Intake(std::string planeName)
...
+//Outbound
 std::vector<double> ReadOilTempsA(HANDLE hProcess, LPVOID codeCaveAddress, std::string planeName)
```
Comment (itsjustdel) [inline Il2ServerAppLateNight/Il2ServerAppLateNight/OilTemp.cpp:80]: Why not just call it ReadTempOilBound

File: Il2ServerAppLateNight/Il2ServerAppLateNight/OilTemp.cpp [hunk 3]
```diff
@@ -24,36 +88,82 @@ std::vector<double> ReadOilTempsA(HANDLE hProcess, LPVOID codeCaveAddress, std::
...
+		values[3] = values[1];
+		values[1] = values[2];
+		values[2] = values[3];
 	}
 
 	return values;
 }
 
-std::vector<double> ReadOilTempsB(HANDLE hProcess, LPVOID codeCaveAddress, std::string planeName)
+//Inbound
```
Comment (itsjustdel) [inline Il2ServerAppLateNight/Il2ServerAppLateNight/OilTemp.cpp:165]: ReadTempOilIBound?

File: Il2ServerAppLateNight/Il2ServerAppLateNight/src/WaterTemp.cpp [hunk 4]
```diff
@@ -18,7 +19,26 @@ std::vector<double> ReadWaterTemps(HANDLE hProcess, LPVOID codeCaveAddress)
 
 		//pointer +178 is offset for water temp in kelvin
 		ReadProcessMemory(hProcess, (LPCVOID)((uintptr_t)targetAddress + 0x178), &rawData, sizeof(double), 0);
+
 		values[i] = *reinterpret_cast<double*>(rawData);
+
+		//Bf 109 K4 Has it's own way to calc water temp
+		std::string v = "Bf 109 K-4";
```
Comment (itsjustdel) [inline Il2ServerAppLateNight/Il2ServerAppLateNight/src/WaterTemp.cpp:26]: Once helper functions for plane string comparison are refactored, use `isBF109K$`

File: Il2ServerAppLateNight/Il2ServerAppLateNight/OilTemp.cpp (hunk 3 above)
Comment (erikesund) [reply Il2ServerAppLateNight/Il2ServerAppLateNight/OilTemp.cpp:165]: I'd suggest "ReadTempOilBound" instead

File: Il2ServerAppLateNight/Il2ServerAppLateNight/OilTemp.cpp (hunk 1 above)
Comment (erikesund) [reply Il2ServerAppLateNight/Il2ServerAppLateNight/OilTemp.cpp:78]: That would definitely make things tidier

File: Il2ServerAppLateNight/Il2ServerAppLateNight/OilTemp.cpp (hunk 3 above)
Comment (itsjustdel) [reply Il2ServerAppLateNight/Il2ServerAppLateNight/OilTemp.cpp:165]: It really makes no sense that suggestion, who's idea was that?

**Pull request merged**